File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int n = nums.length;
        int rs,ls=0 ,total= 0;

        for (int num:nums){
            total+=num;
        }
        
        for(int i=0;i<n;i++){
            rs=total-ls-nums[i];
        
        if(ls==rs) return i;


        ls+=nums[i];
        }
        return -1;
    }
}

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        suffix=[0]*len(nums)
        prefix=[0]*len(nums)

        for i in range(1,len(nums)):
            prefix[i]=prefix[i-1]+nums[i-1]

        for i in range(len(nums)-2,-1,-1):
            suffix[i]=suffix[i+1]+nums[i+1]


            
        for i in range(len(nums)):
            if suffix[i]==prefix[i]:
                return i
            
        return -1
